src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1341,5 +1346,51 @@ private Task<bool> InternalReceiveResponseHeadersAsync(WinHttpRequestState state
 
             return state.TcsReceiveResponseHeaders.Task;
         }
+
+        private static Guid LogRequest(HttpRequestMessage request)
+        {
+            if (_diagnosticSource.IsEnabled("System.Net.Http.Request"))
+            {
+                Guid loggingRequestId = Guid.NewGuid();
+                long timestamp = Stopwatch.GetTimestamp();
+
+                _diagnosticSource.Write(
+                    "System.Net.Http.Request",
+                    new
+                    {
+                        Request = request,
```

Comment:
We have discussed this and that is side effect that we where prepared to accept. It is up to consumers to be responsible citizens and be responsible with what they done. There are logging use cases where mutations might occur (i.e. adding a header) but this isn't the average use case. @vancem